ngx_http_zip_module.c

```diff
@@ -220,6 +190,7 @@ ngx_http_zip_main_request_header_filter(ngx_http_request_t *r)
     ngx_log_debug0(NGX_LOG_DEBUG_HTTP, r->connection->log, 0, "mod_zip: X-Archive-Files found");
 
     if ((ctx = ngx_pcalloc(r->pool, sizeof(ngx_http_zip_ctx_t))) == NULL 
+        || ngx_array_init(&ctx->unparsed_request, r->pool, 64 * 1024, 1) == NGX_ERROR
```

Comment:
Can't we use `Content-Length` header to know size of buffer to allocate?

Reply:
It might not be available if the input is being streamed.

(Could check if it's available and use it in that case - though not sure it's worth the extra complexity.)